PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp

```diff
@@ -2414,9 +2414,9 @@ void ScintillaEditView::convertSelectedTextTo(bool Case)
 		for (int i = 0 ; i < nbChar ; ++i)
 		{
 			if (Case == UPPERCASE)
-				selectedStrW[i] = (WCHAR)::CharUpperW((LPWSTR)selectedStrW[i]);
+				selectedStrW[i] = (WCHAR)(UINT_PTR)::CharUpperW((LPWSTR)selectedStrW[i]);
 			else
-				selectedStrW[i] = (WCHAR)::CharLowerW((LPWSTR)selectedStrW[i]);
+				selectedStrW[i] = (WCHAR)(UINT_PTR)::CharLowerW((LPWSTR)selectedStrW[i]);

```

Comment:
Cast is a small fix for small problem.
I think the big change is not necessary at this point.
Your patch is a good one, you should propose it as a pull request and in your pull request use better replacement fro this API :)

I can do that, and then all nasty part is hidden from the usage.

```cpp
// Wrap CharUpper with cleaner interface
wchar_t MakeCharUpper(wchar_t c)
{
  return (wchar_t)(UINT_PTR)CharUpper((LPWSTR)c);
}
```